### PR TITLE
feature/template_switches_config

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Template Switch config endpoint

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

**Comment by satkunas** [comment]
@jrouzierinverse PATCHing an existing resource blanks all the `type`(s) to "". Likely something missed during key renaming from `name` to `type`.

```json
{
    "item": {
        "acceptRole": [{
            "type": "",
            "value": "$role"
        }],
        "acceptVlan": [{
            "type": "",
            "value": "6"
        }, {
            "type": "",
            "value": "13"
        }, {
            "type": "",
            "value": "$vlan"
        }],
        "coa": [],
        "description": "Standard Switch",
        "disconnect": [{
            "type": "",
            "value": "${macToEUI48($mac)}"
        }, {
            "type": "",
            "value": "$disconnectIp"
        }],
        "id": "PacketFence::Standard",
        "not_deletable": true,
        "not_sortable": false,
        "radiusDisconnect": "disconnect",
        "reject": [{
            "type": "",
            "value": "This node is not allowed to use this service"
        }],
        "voip": [{
            "type": "",
            "value": "6"
        }, {
            "type": "",
            "value": "13"
        }, {
            "type": "",
            "value": "$switch._voiceVlan"
        }]
    },
    "status": 200
}
```

**Comment by jrouzierinverse** [comment]
@julsemaan can you review this, please?
I need this for pfacct.

File: go/caddy/api/api.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		Action:     setup,
+	})
+}
+
+type APIHandler struct {
+	Next   httpserver.Handler
+	router *httprouter.Router
+}
+
+type RadiusAttributeValue struct {
```
Comment (julsemaan) [inline go/caddy/api/api.go]: Just split the RADIUS stuff in a separate file

File: go/caddy/api/api_test.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package api
+
+import (
+	"testing"
+)
+
+func TestSetup(t *testing.T) {
```
Comment (julsemaan) [inline go/caddy/api/api_test.go]: Either implement it or remove this file

**Review by julsemaan — COMMENTED** [review]
I only reviewed the Perl code, @cgx can probably check the JS code

File: html/pfappserver/lib/pfappserver/Form/Config/TemplateSwitch.pm [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+);
+
+has_field 'description' => (
+    type     => 'Text',
+    required => 1,
+);
+
+has_field 'radiusDisconnect' => (
+    type    => 'Select',
+    label   => 'Radius Disconnect Method',
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/TemplateSwitch.pm]: s/Radius/RADIUS/

File: html/pfappserver/root/static.alt/src/store/modules/config.js [hunk 4]
```diff
@@ -1257,6 +1265,20 @@ const actions = {
...
+  getSwitchTemplates: ({ state, getters, commit }) => {
+    if (getters.isLoadingSwitchTemplates) {
+      return Promise.resolve(state.switchTemplates)
+    }
+    if (!state.switchTemplates) {
+      commit('SWICTH_TEMPLATES_REQUEST')
+      return api.getSwitchTemplates().then(response => {
+        commit('SWICTH_TEMPLATES_UPDATED', response.data.items)
+        return state.switchTemplates
+      })
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/store/modules/config.js:1283]: Update `state.switchTemplatesStatus` when the API call fails.

**Comment by jrouzierinverse** [comment]
All requested items except the JS stuff @satkunas can you check it out?

**Comment by jrouzierinverse** [comment]
Everything is updated please re-review.

**Comment by cgx** [comment]
All good for the frontend part.

**Pull request merged**